### PR TITLE
Update Visual Studio Version to 17.0

### LINKED ISSUE #6360: 17.0 branding update incomplete
### Issue Description

If I run the script below in a vanilla PowerShell windows, both versions print `16.0`. Should dev17 prints `17.0` ?

### Steps to Reproduce

Project file:
 
```xml
<Project>
  <Target Name="Build">
    <Message Text="Current VS: $(VisualStudioVersion)" Importance="High" />
  </Target>
</Project>
```
 
Script:
 
```powershell
$ErrorActionPreference = 'Stop'

$msbuild16 = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe"
$msbuild17 = "C:\Program Files\Microsoft Visual Studio\17\PreviewPartner\MSBuild\Current\Bin\MSBuild.exe"

echo "Testing dev16 EnvVar VisualStudioVersion" $env:VisualStudioVersion
. $msbuild16 testvs.proj /nologo /v:m /nr:false /bl:test16.binlog
echo "Testing dev17 EnvVar VisualStudioVersion" $env:VisualStudioVersion 
. $msbuild17 testvs.proj /nologo /v:m /nr:false /bl:test17.binlog
```

### Expected Behavior

```text
Testing dev16 EnvVar VisualStudioVersion
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe /nologo /bl:test16.binlog /nr:false /v:m testvs.proj
  Current VS: 16.0
Testing dev17 EnvVar VisualStudioVersion
C:\Program Files\Microsoft Visual Studio\17\PreviewPartner\MSBuild\Current\Bin\MSBuild.exe /nologo /bl:test17.binlog /nr:false /v:m testvs.proj
  Current VS: 17.0
```

### Actual Behavior

```text
Testing dev16 EnvVar VisualStudioVersion
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe /nologo /bl:test16.binlog /nr:false /v:m testvs.proj
  Current VS: 16.0
Testing dev17 EnvVar VisualStudioVersion
C:\Program Files\Microsoft Visual Studio\17\PreviewPartner\MSBuild\Current\Bin\MSBuild.exe /nologo /bl:test17.binlog /nr:false /v:m testvs.proj
  Current VS: 16.0
```

### Analysis

The project file is not declaring any SDK or import. The following props file might be related. Thanks to @davkean for pointing it out.

https://github.com/dotnet/msbuild/blob/9bcc06cbe19ae2482ab18eab90a82fd079b26897/src/Tasks/Microsoft.VisualStudioVersion.v16.Common.props#L16

### Versions & Configurations

```
 . "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe" /nologo /version
16.9.0.16703
```

```
. "C:\Program Files\Microsoft Visual Studio\17\PreviewPartner\MSBuild\Current\Bin\MSBuild.exe" /nologo /version
17.0.0.20804
```

Windows 10 Version 20H2 (19042.928)

### Attach a binlog

[binlogs.zip](https://github.com/dotnet/msbuild/files/6347176/binlogs.zip)

### PR DESCRIPTION
**Pull request by Forgind**
Fixes #6360 

### Context
Branding update didn't hit Visual Studio version

### Changes Made
Fixed that

### Testing
None

### Notes
I messed up on my commit history, but the changes are right. Shouldn't matter if we squash it in.

### CODE REVIEW

**Review by rainersigwald — CHANGES_REQUESTED** [review]
We'll also need a bunch of the stuff from #6336. Let's merge that in, fi to 17.0, and add the new changes.

**Comment by Forgind** [comment]
Yeah—I saw there was a lot that depended on "Current" version. So to clarify, no action item for this PR until the first two steps are complete, and then this will be good to squash?

**Comment by rainersigwald** [comment]
> no action item for this PR until the first two steps are complete, 

correct

> and then this will be good to squash?

No; we'll want to add a couple of 17.0 things on top of Kirill's PR.

**Review by BenVillalobos — APPROVED** [review]

File: src/Tasks/Microsoft.VisualStudioVersion.v17.Common.props [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿<!--
+***********************************************************************************************
+Microsoft.VisualStudio.v16.Common.props
```
Comment (cdmihai) [inline src/Tasks/Microsoft.VisualStudioVersion.v17.Common.props:3]:
```suggestion
Microsoft.VisualStudio.v17.Common.props
```

**Comment by rainersigwald** [comment]
@Forgind can you merge main to 17, then rebase or remerge this? Then we'll need to add a couple more things.

**Comment by Forgind** [comment]
I merged main into this, but I don't think I have permission to push directly to the 17.0 branch, so I'd have to add it via PR.

I made a couple tweaks that looked necessary from 6336. What did I miss?

**Comment by rainersigwald** [comment]
> I merged main into this, but I don't think I have permission to push directly to the 17.0 branch, so I'd have to add it via PR.

Yes, please. As is the diff is very confusing, and the commit history will be a bit of a mess (when we eventually come along and do 18.0 I'd like to have as few commits as possible so we can look back).

**Comment by rainersigwald** [comment]
I'm still finding this diff intensely confusing. Mind squashing this branch into a better form?

```sh-session
❯ git checkout 17.0
Already on '17.0'
Your branch is up to date with 'Forgind/17.0'.

❯ git rev-parse HEAD
fb1d9659863d5e77053d86a23fe9ca06e4575013

❯ git reset --hard upstream/vs17.0

❯ git merge --squash fb1d9659863d5e77053d86a23fe9ca06e4575013
Automatic merge went well; stopped before committing as requested
Squash commit -- not updating HEAD

❯ git commit -am "Additional 17.0 branding/internal updates"
[17.0 8519d595c6] Additional 17.0 branding/internal updates
 10 files changed, 58 insertions(+), 16 deletions(-)
 create mode 100644 src/Tasks/Microsoft.VisualStudioVersion.v17.Common.props
```

**Comment by Forgind** [comment]
I'd neglected to do any of that since main merged into vs17.0. It was also nice to have the commands written out for me so I didn't have to think about it 🙂

File: src/Utilities/ToolLocationHelper.cs [hunk 2]
```diff
@@ -160,15 +160,15 @@ public enum VisualStudioVersion
         Version160,
 
         /// <summary>
-        /// Visual Studio "Dev17"
+        /// Visual Studio (Dev17)
```
Comment (rainersigwald) [inline src/Utilities/ToolLocationHelper.cs:163]: The planned name [is public](https://devblogs.microsoft.com/visualstudio/visual-studio-2022/).

```suggestion
        /// Visual Studio 2022 (Dev17)
```

**Review by rainersigwald — APPROVED** [review]
At last it's understandable! I think GitHub got confused about what the merge-base should be because vs17.0 was moved to a place slightly ahead of your branch. I don't know why that was confusing though . . .

image

File: src/Utilities/ToolLocationHelper.cs (hunk 2 above)
Comment (Forgind) [reply src/Utilities/ToolLocationHelper.cs]: BenVillalobos made that exact comment in KirillOsenkov's PR:
https://github.com/dotnet/msbuild/pull/6336#discussion_r617709753

Somehow it wasn't merged, apparently 😄 

Thanks!

**Review by rainersigwald — CHANGES_REQUESTED** [review]
Please change the VSIX to contain the new VS file:

https://github.com/dotnet/msbuild/blob/d4f6926dfcba701cec3357985b5b890a85e3d8b8/src/Package/MSBuild.VSSetup/files.swr#L24

**Comment by ladipro** [comment]
@Forgind after addressing feedback please consider repointing to `main` which is now becoming our 17.0 development branch.

**Review by rainersigwald — APPROVED** [review]
@marcpopMSFT this is the PR I'd really like in VS 17.0 ASAP.

**Comment by rainersigwald** [comment]
Looks like missing this is breaking some internal C++ tests. @ladipro I'll include you on the mail. We should take this ASAP and get an insertion going so we can let the QB figure out when to take it.

**Pull request merged**